src/hip_memory.cpp

```diff
@@ -410,6 +410,113 @@ extern void getChannelOrderAndType(const hipChannelFormatDesc& desc,
                             hsa_ext_image_channel_order_t& channelOrder,
                             hsa_ext_image_channel_type_t& channelType);
 
+hipError_t hipArrayCreate ( hipArray** array, const HIP_ARRAY_DESCRIPTOR* pAllocateArray )
+{
+    HIP_INIT_SPECIAL_API((TRACE_MEM), array, pAllocateArray);
+    HIP_SET_DEVICE();
+    hipError_t  hip_status = hipSuccess;
+    if(pAllocateArray->width >0) {
+		auto ctx = ihipGetTlsDefaultCtx();
+
+		*array = (hipArray*)malloc(sizeof(hipArray));
+        array[0]->drvDesc = *pAllocateArray;
+        array[0]->width = pAllocateArray->width;
+        array[0]->height = pAllocateArray->height;
+        array[0]->isDrv = true;
+        void ** ptr = &array[0]->data;
+	    if (ctx) {
+			const unsigned am_flags = 0;
+			size_t size = pAllocateArray->width;
+			if(pAllocateArray->height > 0) {
+				size = size * pAllocateArray->height;
+			}
+            hsa_ext_image_channel_type_t channelType;
+			size_t allocSize = 0;
+			switch(pAllocateArray->format) {
+				case HIP_AD_FORMAT_UNSIGNED_INT8:
+					allocSize = size * sizeof(unsigned char);
+					channelType = HSA_EXT_IMAGE_CHANNEL_TYPE_UNSIGNED_INT8;
+					break;
+				case HIP_AD_FORMAT_UNSIGNED_INT16:
+					allocSize = size * sizeof(unsigned short);
```

Comment:
int16_t

Reply:
And more below - since the enumeration have specific size we should use the standard types with size encoded for these conversions.